src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,29 +72,100 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
-
-  // to avoid bug undefined url when user change number of images
-  useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
-    }
   }, [numberOfImgs]);
 
   return (
     <ImageGrid
       height={imagesGridHeight}
-      width={imagesGridWidth}
+      maxWidth={imagesGridMaxWidth}
       numberOfImgs={numberOfImgs}
       row={rowCol.row}
       col={rowCol.col}
     >
-      {images.map((img, index) => (
-        <ImageWrap>
-          <ImageItem src={img} />
-        </ImageWrap>
-      ))}
+      {numberOfImgs <= 5 &&
+        images
+          .slice(0, numberOfImgs)
+          .map((photo: UnsplashPhotoFortmat, index) => (
+            <ImageWrap>
+              <ImageItem
+                src={photo.url || "default.jpg"}
+                alt={photo?.alt || "photo"}
+              />
+            </ImageWrap>
+          ))}
+
+      {numberOfImgs > 5 &&
+        images.slice(0, 5).map((photo: UnsplashPhotoFortmat, index) => (
+          <ImageWrap style={{ position: "relative" }}>
+            <ImageItem
+              src={photo.url || "default.jpg"}
+              alt={photo?.alt || "photo"}
+            />
+            {index === 4 && (
+              <p
+                style={{
+                  position: "absolute",
+                  top: "50%",
+                  left: "50%",
+                  transform: "translate(-50%, -50%)",
+                  color: "white",
+                  fontSize: "1.5rem",
+                }}
+              >
+                + {numberOfImgs - 5}
+              </p>
+            )}
+          </ImageWrap>
+        ))}
     </ImageGrid>
   );
 };
 export default ImageGridComponent;
+
+/*----------------------EMOTION STYLED---------------------- */
+
+const ImageItem = styled.img`
+  object-fit: cover;
+  object-position: "center";
+  min-width: 100%;
+  height: 100%;
+`;
+
+let ImageWrap = styled.div<StyledImageWrap>((props) => ({
```

Comment:
this style does not use props, you don't need this function, can just use template literal